openssl-dynamic/src/main/c/ssl_private.h

```diff
@@ -125,7 +125,7 @@
 
 #define MAX_ALPN_NPN_PROTO_SIZE 65535
 
-static const char* UNKNOWN_AUTH_METHOD = "UNKNOWN";
+#define UNKNOWN_AUTH_METHOD "UNKNOWN"
```

Comment:
we shouldn't have to loose the type safety. Lets do this instead (note I added the `TCN_` prefix to provide more scope and reduce the chance of collisions):

```diff
diff --git a/openssl-dynamic/src/main/c/ssl_private.h b/openssl-dynamic/src/main/c/ssl_private.h
index 1c9e4c4..f4b1338 100644
--- a/openssl-dynamic/src/main/c/ssl_private.h
+++ b/openssl-dynamic/src/main/c/ssl_private.h
@@ -125,7 +125,7 @@
 
 #define MAX_ALPN_NPN_PROTO_SIZE 65535
 
-#define UNKNOWN_AUTH_METHOD "UNKNOWN"
+extern const char* TCN_UNKNOWN_AUTH_METHOD;
 
 /* ECC: make sure we have at least 1.0.0 */
 #if !defined(OPENSSL_NO_EC) && defined(TLSEXT_ECPOINTFORMAT_uncompressed)
diff --git a/openssl-dynamic/src/main/c/sslcontext.c b/openssl-dynamic/src/main/c/sslcontext.c
index 2fa688d..bdcf38e 100644
--- a/openssl-dynamic/src/main/c/sslcontext.c
+++ b/openssl-dynamic/src/main/c/sslcontext.c
@@ -1183,7 +1183,7 @@ static const char* authentication_method(const SSL* ssl) {
             ciphers = SSL_get_ciphers(ssl);
             if (ciphers == NULL || sk_SSL_CIPHER_num(ciphers) <= 0) {
                 // No cipher available so return UNKNOWN.
-                return UNKNOWN_AUTH_METHOD;
+                return TCN_UNKNOWN_AUTH_METHOD;
             }
             return SSL_cipher_authentication_method(sk_SSL_CIPHER_value(ciphers, 0));
         }
diff --git a/openssl-dynamic/src/main/c/sslutils.c b/openssl-dynamic/src/main/c/sslutils.c
index 77cbf94..e5eb501 100644
--- a/openssl-dynamic/src/main/c/sslutils.c
+++ b/openssl-dynamic/src/main/c/sslutils.c
@@ -69,6 +69,8 @@
 #define SSL_aGOST94     0x00000100L /* GOST R 34.10-94 signature auth */
 #define SSL_aGOST01     0x00000200L /* GOST R 34.10-2001 signature auth */
 
+const char* TCN_UNKNOWN_AUTH_METHOD = "UNKNOWN";
+
 /* OpenSSL end */
 
 /*
@@ -96,7 +98,7 @@ const char* SSL_cipher_authentication_method(const SSL_CIPHER* cipher){
         case SSL_aNULL:
             return SSL_TXT_DH "_anon";
         default:
-            return UNKNOWN_AUTH_METHOD;
+            return TCN_UNKNOWN_AUTH_METHOD;
             }
     case SSL_kKRB5:
         return SSL_TXT_KRB5;
@@ -114,10 +116,10 @@ const char* SSL_cipher_authentication_method(const SSL_CIPHER* cipher){
         case SSL_aNULL:
             return SSL_TXT_ECDH "_anon";
         default:
-            return UNKNOWN_AUTH_METHOD;
+            return TCN_UNKNOWN_AUTH_METHOD;
             }
     default:
-        return UNKNOWN_AUTH_METHOD;
+        return TCN_UNKNOWN_AUTH_METHOD;
     }
 #else
     return SSL_CIPHER_get_kx_name(cipher);
```